akkeeper/src/main/scala/akkeeper/config/RichConfig.scala

```diff
@@ -15,102 +15,133 @@
  */
 package akkeeper.config
 
-import akkeeper.api.DeployContainer
+import java.time.{Duration => JavaDuration}
+import java.util.concurrent.TimeUnit
+
+import akka.util.Timeout
 import akkeeper.common.ContainerDefinition
 import akkeeper.master.service.MasterService
 import akkeeper.storage.zookeeper.ZookeeperClientConfig
 import com.typesafe.config.{Config, ConfigValueFactory}
 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs.{FileSystem, Path}
+
 import scala.collection.JavaConverters._
+import scala.collection.mutable
+import scala.concurrent.duration._
 
 private[akkeeper] final class RichConfig(config: Config) {
 
-  def withMasterRole: Config = {
-    config.withValue("akka.cluster.roles",
-      ConfigValueFactory.fromIterable(Seq(MasterService.MasterServiceName).asJava))
+  implicit def javaDuration2ScalaDuration(value: JavaDuration): FiniteDuration = {
+    Duration.fromNanos(value.toNanos)
   }
 
-  def withMasterPort: Config = {
-    config.withValue("akka.remote.netty.tcp.port",
-      ConfigValueFactory.fromAnyRef(config.getInt("akkeeper.akka.port")))
-  }
+  private def akkeeperConfig: Config = config.getConfig("akkeeper")
 
-  def withPrincipalAndKeytab(principal: String, keytab: String): Config = {
-    config
-      .withValue("akka.kerberos.principal", ConfigValueFactory.fromAnyRef(principal))
-      .withValue("akka.kerberos.keytab", ConfigValueFactory.fromAnyRef(keytab))
-  }
+  // Top-level accessors
+  private def akkaConfig: Config = akkeeperConfig.getConfig("akka")
+  private def monitoringConfig: Config = akkeeperConfig.getConfig("monitoring")
+  private def launcherConfig: Config = akkeeperConfig.getConfig("launcher")
+  private def zookeeperConfig: Config = akkeeperConfig.getConfig("zookeeper")
+  private def restConfig: Config = akkeeperConfig.getConfig("api.rest")
+  private def kerberosConfig: Config = akkeeperConfig.getConfig("kerberos")
+  private def yarnConfig: Config = akkeeperConfig.getConfig("yarn")
+  private def yarnMaster: Config = yarnConfig.getConfig("master")
```

Comment:
Please remove the `yarn` prefix since the `master` settings are not related to YARN. Potentially Akkeeper can be used with systems other than YARN in the future.

Reply:
The `Config` suffix is also missing which makes this name inconsistent with others.